### PR TITLE
Add suite-native PR process tools

### PR DESCRIPTION
**Pull request by arthexis**
## Summary

Adds suite-native PR process commands and bundles the supporting operator-framework skill updates so Windows and WSL Codex sessions can share the same PR oversight workflow.

## Commands Added

- `.venv/bin/python manage.py pr_oversee --repo arthexis/arthexis review-batch --pr <number> --format markdown`
- `.venv/bin/python manage.py pr_oversee --repo arthexis/arthexis domain-preflight --pr <number> --format markdown`
- `.venv/bin/python manage.py pr_oversee --repo arthexis/arthexis report closed --since 8h`
- `.venv/bin/python manage.py pr_oversee --repo arthexis/arthexis monitor --pr <number> --compact`
- `.venv/bin/python manage.py gh_monitor status --limit 10`
- `python3 "${CODEX_HOME:-$HOME/.codex}/skills/arthexis-review-attend/scripts/patchwork_prepare.py" --write --format markdown`
- `python3 "${CODEX_HOME:-$HOME/.codex}/skills/arthexis-review-attend/scripts/review_thread.py" list --repo arthexis/arthexis --pr <number> --unresolved --format compact`
- `python3 "${CODEX_HOME:-$HOME/.codex}/skills/arthexis-review-attend/scripts/review_thread.py" gate --repo arthexis/arthexis --pr <number>`
- `python3 "${CODEX_HOME:-$HOME/.codex}/skills/arthexis-review-attend/scripts/review_thread.py" reply-resolve-many --body-file ./thread-replies.json`
- `python3 "${CODEX_HOME:-$HOME/.codex}/skills/arthexis-review-attend/scripts/review_thread.py" update-pr-body --repo arthexis/arthexis --pr <number> --body-file ./pr-body.md`

## Details

- Adds review-thread batch summaries, domain-specific preflight checks, compact monitor output, and recent closed-PR reporting to `pr_oversee`.
- Adds a read-only `gh_monitor status` dashboard for queued/active assignment work.
- Refreshes the bundled `operator-framework-core` PR skills to prefer suite-native commands, WSL patchwork paths, and Codex-home fallback shell snippets.
- Adds a portable packaged `patchwork_prepare.py` helper for linking patchwork venvs and creating isolated SQLite test paths.
- Extends review-thread helper tooling with compact listing, review-gate summaries, batch reply/resolve, and PR body update support.

## Validation

- `.venv/bin/ruff check apps/repos/pr_oversee/service.py apps/repos/management/commands/pr_oversee.py apps/repos/github_monitor.py apps/repos/management/commands/gh_monitor.py apps/repos/tests/test_pr_oversee.py apps/repos/tests/test_gh_monitor.py apps/skills/packages/operator-framework-core/skills/arthexis-review-attend/scripts/review_thread.py apps/skills/packages/operator-framework-core/skills/arthexis-review-attend/scripts/patchwork_prepare.py`
- `.venv/bin/ruff check apps/skills/tests/test_package_services.py`
- `git diff --check`
- `ARTHEXIS_SQLITE_TEST_PATH=/home/arthe/patchwork-linux/arthexis-pr-process-tools/work/test-db/test-pr-oversee-final-2.sqlite3 .venv/bin/python manage.py test run -- apps/repos/tests/test_pr_oversee.py`
- `ARTHEXIS_SQLITE_TEST_PATH=/home/arthe/patchwork-linux/arthexis-pr-process-tools/work/test-db/test-gh-monitor-final.sqlite3 .venv/bin/python manage.py test run -- apps/repos/tests/test_gh_monitor.py`
- `ARTHEXIS_SQLITE_TEST_PATH=/home/arthe/patchwork-linux/arthexis-pr-process-tools/work/test-db/test-skills-package-final-2.sqlite3 .venv/bin/python manage.py test run -- apps/skills/tests/test_operator_framework_core_scripts.py apps/skills/tests/test_package_services.py`
- `ARTHEXIS_SQLITE_TEST_PATH=/home/arthe/patchwork-linux/arthexis-pr-process-tools/work/test-db/test-manage-check.sqlite3 .venv/bin/python manage.py check --fail-level ERROR`
- `ARTHEXIS_SQLITE_TEST_PATH=/home/arthe/patchwork-linux/arthexis-pr-process-tools/work/test-db/test-migrations-check.sqlite3 .venv/bin/python manage.py migrations check`
- `ARTHEXIS_SQLITE_TEST_PATH=/home/arthe/patchwork-linux/arthexis-pr-process-tools/work/test-db/test-package-dry-run.sqlite3 .venv/bin/python manage.py codex_skill_packages import --package apps/skills/packages/operator-framework-core --dry-run`

## References

Refs #8739. Related to #8731, #8751, and #8748.

# PR Summary: Suite-Native PR Process Tools

## Overview
This PR consolidates pull request oversight workflows into the main application, introducing suite-native PR management commands and helper scripts that enable Windows and WSL Codex sessions to share the same PR process. The contribution includes ten new management commands and helper scripts for review analysis, preflight validation, closed-PR reporting, and review thread operations.

### New Management Commands and Services
- **`gh_monitor status`**: A read-only dashboard command that displays GitHub monitor readiness info, per-item status counts, per-task queue/active counts, and a prioritized list of monitor items (ordered by status priority: ACTIVE before QUEUED).
- **`pr_oversee review-batch`**: Batches and summarizes review threads by severity and path, with optional inclusion of resolved threads and markdown formatting support.
- **`pr_oversee domain-preflight`**: Performs domain-specific preflight checks by matching PR titles, bodies, and changed file paths against domain rules, assembling targeted validation commands and assigning risk levels.
- **`pr_oversee report closed`**: Reports closed PRs using human-duration syntax (e.g., `--since 8h`, `--since 2d`).
- **`pr_oversee monitor --compact`**: Compacts monitor output into a derivable outcome state (merged/manual/ready/waiting/blocked/complete/inspect) with summarized check counts and blockers.

### Extended Review Thread Helpers
The `review_thread.py` script now provides:
- Pagination-aware thread fetching with compact listing and text truncation
- Review gate summaries that compute thread counts and derive gate status based on unresolved threads and review decisions
- Batch reply/resolve operations via `reply-resolve-many` (reading JSON input)
- PR body updates via `update-pr-body` REST PATCH command
- GraphQL and REST API helpers for authentication, patching, and JSON execution

### Patchwork Preparation Utility
A new `patchwork_prepare.py` helper:
- Computes isolated SQLite test DB paths for patchwork checkouts
- Links patchwork checkouts to a source checkout's `.venv`
- Handles Windows symlink/junction creation with fallback support
- Outputs structured payload with resolved paths and environment-variable command strings

### Domain-Aware PR Analysis
New PR oversight service features:
- Review severity detection from PR bodies (P0–P3 heuristics)
- Domain preflight rules matching against file paths and PR content
- Check state enumeration with pending/failing/superseded/passing counts
- Outcome derivation for monitor result compaction

## Operator-Framework-Core Skill Package Updates
Documentation and manifests for three skills were refreshed:
- **arthexis-review-attend**: Updated with POSIX/Bash-style commands, patchwork preparation steps, and WSL device assignment protocols
- **arthexis-pr-oversee**: Standardized local/remote command usage, added device assignment and local development gate sections, updated priority-helper workflows
- **arthexis-pr-oversee-monitor**: Added `--compact` variants, `--expected-head-sha` guarded merge examples, and Windows Python path usage guidance

The manifest for `arthexis-review-attend` now includes `patchwork_prepare.py` in its file list.

## Testing
Comprehensive test coverage was added across multiple modules:
- `test_gh_monitor.py`: Validates status payload structure and item prioritization
- `test_pr_oversee.py`: Tests review batch ordering, domain preflight risk detection, compact monitor formatting, CLI parsing, and duration parsing
- `test_operator_framework_core_scripts.py`: Validates thread pagination and Windows path compilation
- `test_package_services.py`: Updated expectations for operator-framework-core skill import (4 files instead of 3)

Tests focus on practical validation of command behavior and output formatting rather than edge-case coverage.

## Code Quality and Validation
The PR includes:
- Linting with ruff
- Test suite runs with isolated SQLite databases for pr_oversee, gh_monitor, and operator-framework-core scripts
- Django system checks and migrations validation
- Dry-run of operator-framework-core skill package import
- Code addressing Sonar findings, pagination improvements, and REST API usage for PR body updates

## Platform Compatibility
Updates include WSL patchwork paths, Windows Python path conventions, Windows junction fallback support, and Codex-home fallback shell snippets (`${CODEX_HOME:-$HOME/.codex}`) for improved cross-platform compatibility.

### CODE REVIEW

File: apps/repos/github_monitor.py [hunk 1]
```diff
@@ -621,6 +621,77 @@ def evaluate_readiness() -> dict[str, Any]:
...
+        key=lambda item: (
+            status_order.get(item.status, 99),
+            item.queued_at,
+            item.pk,
+        )
+    )
+    item_counts = {
+        status: GitHubMonitorItem.objects.filter(status=status).count()
+        for status in GitHubMonitorItem.Status.values
+    }
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/repos/github_monitor.py:676]: Addressed in 66d4f5cfa by aggregating GitHubMonitorItem status counts in one query. Validation: ruff, py_compile, git diff --check, and focused tests passed.

File: apps/repos/github_monitor.py [hunk 2]
```diff
@@ -621,6 +621,77 @@ def evaluate_readiness() -> dict[str, Any]:
...
+            "repository": task.repository.slug,
+            "targetType": task.target_type,
+            "codexCommand": task.codex_command,
+            "queued": task.items.filter(status=GitHubMonitorItem.Status.QUEUED).count(),
+            "active": task.items.filter(status=GitHubMonitorItem.Status.ACTIVE).count(),
+        }
+        for task in GitHubMonitorTask.objects.select_related("repository").order_by(
+            "name"
+        )
+    ]
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/repos/github_monitor.py:700]: Addressed in 66d4f5cfa by annotating queued and active item counts on the GitHubMonitorTask queryset. Validation: ruff, py_compile, git diff --check, and focused tests passed.

File: apps/skills/packages/operator-framework-core/skills/arthexis-review-attend/scripts/patchwork_prepare.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+def create_venv_link(link_path: Path, target_path: Path) -> tuple[str, str]:
+    if os.name == "nt":
+        completed = subprocess.run(
+            ["cmd", "/c", "mklink", "/J", str(link_path), str(target_path)],
+            check=False,
+            capture_output=True,
+            encoding="utf-8",
+            errors="replace",
+            text=True,
+        )
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/skills/packages/operator-framework-core/skills/arthexis-review-attend/scripts/patchwork_prepare.py]: Addressed in 66d4f5cfa by using Path.symlink_to first and retaining a non-shell Windows junction fallback when symlink creation is unavailable. Validation: ruff, py_compile, git diff --check, and focused tests passed.

File: apps/skills/packages/operator-framework-core/skills/arthexis-review-attend/scripts/patchwork_prepare.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+import argparse
+import json
+import os
+import re
+import sys
+from pathlib import Path
+from typing import Any
+
+DEFAULT_SOURCE_CHECKOUT = "[CONF.BASE_DIR]"
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/skills/packages/operator-framework-core/skills/arthexis-review-attend/scripts/patchwork_prepare.py]: Implemented in 70c699021: patchwork_prepare now computes the source checkout at runtime from ARTHEXIS_CHECKOUT or ~/arthexis instead of embedding the substituted [CONF.BASE_DIR] token in Python source. Validation: ruff passed for the changed files; py_compile and a runtime --format json helper invocation passed; apps/repos/tests/test_pr_oversee.py passed, 76 tests.

File: apps/repos/pr_oversee/service.py [hunk 5]
```diff
@@ -403,6 +495,152 @@ def comments(self, number: int, *, unresolved_only: bool = False) -> dict[str, A
...
+        ]
+        touched_lower = touched_paths.casefold()
+        if "apps/cards/" in touched_paths or "rfid" in touched_lower:
+            validation.append("python manage.py test apps.cards")
+        if "apps/imager/" in touched_paths:
+            validation.append("python manage.py test apps.imager")
+        if "apps/nodes/" in touched_paths:
+            validation.append("python manage.py test apps.nodes")
+        if "apps/nmcli/" in touched_paths or "ap_portal" in touched_lower:
+            validation.append("python manage.py test apps.nmcli")
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/repos/pr_oversee/service.py]: Implemented in 70c699021: domain-preflight now emits .venv/bin/python manage.py test run -- <target> commands for app-specific validations, and the domain-preflight test expectations were updated. Validation: ruff passed for the changed files; apps/repos/tests/test_pr_oversee.py passed, 76 tests.

File: apps/skills/packages/operator-framework-core/skills/arthexis-review-attend/scripts/patchwork_prepare.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/skills/packages/operator-framework-core/skills/arthexis-review-attend/scripts/patchwork_prepare.py]: Addressed in 70c699021: patchwork_prepare now computes the default source checkout at runtime instead of materializing [CONF.BASE_DIR] into a Python string literal. Validation: ruff on touched files, py_compile for patchwork_prepare.py, operator-framework package dry-run, and apps/repos/tests/test_pr_oversee.py with 76 passed.

File: apps/repos/pr_oversee/service.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/repos/pr_oversee/service.py]: Addressed in 70c699021: domain-preflight now emits the canonical .venv/bin/python manage.py test run -- <target> form for app-specific validation commands. Validation: ruff on touched files, py_compile for patchwork_prepare.py, operator-framework package dry-run, and apps/repos/tests/test_pr_oversee.py with 76 passed.

File: apps/skills/packages/operator-framework-core/skills/arthexis-review-attend/scripts/patchwork_prepare.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/skills/packages/operator-framework-core/skills/arthexis-review-attend/scripts/patchwork_prepare.py]: Addressed in ed37e29f6 by preserving the suite materialized checkout path as a raw default while allowing ARTHEXIS_CHECKOUT and ARTHEXIS_BASE_DIR overrides. Validation: ruff, py_compile, git diff --check, and focused test modules passed (90 passed).

File: apps/repos/github_monitor.py [hunk 6]
```diff
@@ -621,6 +622,88 @@ def evaluate_readiness() -> dict[str, Any]:
...
+        GitHubMonitorItem.Status.QUEUED: 1,
+        GitHubMonitorItem.Status.FAILED: 2,
+        GitHubMonitorItem.Status.TIMED_OUT: 3,
+        GitHubMonitorItem.Status.CLOSED: 4,
+        GitHubMonitorItem.Status.COMPLETED: 5,
+        GitHubMonitorItem.Status.DISMISSED: 6,
+    }
+    items = list(
+        GitHubMonitorItem.objects.select_related("task", "task__repository")
+        .order_by("-queued_at", "-id")[: max(int(limit or 0), 0)]
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/repos/github_monitor.py]: Implemented in db73ef93e: gh_monitor status now annotates status priority in the queryset and orders active/queued rows before applying --limit, so older active items are not hidden by newer completed rows. Added a regression test for active-before-limit behavior. Validation: ruff passed for the changed files; focused gh_monitor status tests passed.

File: apps/skills/packages/operator-framework-core/skills/arthexis-review-attend/scripts/review_thread.py [hunk 7]
```diff
@@ -164,8 +278,115 @@ def reply_summary(args: argparse.Namespace) -> None:
...
+            args.repo,
+            "--json",
+            "headRefOid,reviewDecision,latestReviews,statusCheckRollup",
+        ]
+    )
+    threads, error = fetch_threads(args.repo, args.pr)
+    if error:
+        print(json.dumps({"error": error}, indent=2))
+        raise SystemExit(1)
+    summary = summarize_review_gate(pr, threads)
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/skills/packages/operator-framework-core/skills/arthexis-review-attend/scripts/review_thread.py:386]: Implemented in db73ef93e: the packaged review_thread.py fetch_threads helper now follows reviewThreads pageInfo until all pages are loaded before list/gate summaries are computed. Added a two-page fetch regression test. Validation: ruff passed for the changed files; focused operator-framework review_thread tests passed.

[the remark this answers is not in the document]
Comment (arthexis) [reply apps/skills/packages/operator-framework-core/skills/arthexis-review-attend/scripts/review_thread.py:386]: Addressed in db73ef93e by paging reviewThreads with pageInfo/endCursor before summarizing the packaged review gate. Added a regression test that fetch_threads follows the next-page cursor. Validation: ruff and py_compile passed for the touched files; focused Django modules passed (59 passed).

File: apps/repos/pr_oversee/service.py [hunk 8]
```diff
@@ -1794,3 +2032,74 @@ def _monitor_result(
...
+        )
+        merged = bool(_coerce_mapping(merge_action.get("result")).get("merged"))
+        blockers = [str(item) for item in _coerce_list(readiness.get("blockers"))]
+        state = str(readiness.get("state") or pr.get("state") or "")
+        ready = bool(readiness.get("ready"))
+        if state == "MERGED" or merged:
+            outcome = "merged"
+            blockers = []
+        elif ready and not blockers:
+            outcome = "ready"
```
[the remark this answers is not in the document]
Comment (arthexis) [reply apps/repos/pr_oversee/service.py:2089]: Addressed in 0fac46b1c by making compact_monitor_result report outcome=manual before ready when manualDecisionRequired is set. Added a regression test for a ready gate with a manual local-validation stop. Validation: ruff, py_compile, git diff --check, and focused Django modules passed (92 passed).
